### ISSUE TITLE
Always Overloaded

### ISSUE THREAD

**Issue by MrSnake0208** [mornedhels/enshrouded-server#7]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Used image version

f1b97f0461172c352038b2a5f6ff424f2f73dce757d4878b952fdc9ae2524518 

### What you expected to see, versus what you actually saw

After my server is started, as soon as someone enters, its latency becomes extremely high, and it shows server overload. However, in reality, it only occupies 3-4GB of memory and some CPU. I saw someone on Reddit with a similar situation, and below is someone's reply:

> EDIT: The wonderful Vapok in the official Enshrouded Discord gave me a configuration that uses Proton instead of Wine made by the amazing gOOvER... what a difference. All lag gone, CPU usage sliced in half, RAM usage dropped, no more warnings in-game.
[Here's a link to the GitHub that has the PROTON egg](https://github.com/gOOvER/own-pterodactyl-eggs/blob/main/steamcmd/enshrouded/egg-enshrouded-proton.json) for those interested.

Could you please take a look?

**Comment by cp-fabian-pittroff** [comment]
Hey, I'm already on it. Will probably be a alternative tag to use.

I try to cleanup my scripts and test it in a few hours. I also will need some help from the community to see the load difference hopefully. I will ask for that here [Discord](https://discord.com/channels/1046840540550942720/1199773479914979458).

**Comment by MrSnake0208** [comment]
> Hey, I'm already on it. Will probably be a alternative tag to use.
> 
> 
> 
> I try to cleanup my scripts and test it in a few hours. I also will need some help from the community to see the load difference hopefully. I will ask for that here [Discord](https://discord.com/channels/1046840540550942720/1199773479914979458).

Thx

**Comment by cp-fabian-pittroff** [comment]
I keep the issue open, you can try out the new dev image which uses proton (`dev-proton`).

**Comment by MrSnake0208** [comment]
> I keep the issue open, you can try out the new dev image which uses proton (`dev-proton`).我仍在处理这个问题，你可以试试使用质子的新开发镜像 ( `dev-proton` )。
This is my log, something may be wrong?
[_enshrouded_logs.txt](https://github.com/mornedhels/enshrouded-server/files/14073017/_enshrouded_logs.txt)

**Comment by cp-fabian-pittroff** [comment]
ah, jeah I stumbled over the same problem. I will fix that in a future update. For now you have to force an update.

```bash
docker compose exec enshrouded supervisorctl start enshrouded-force-update
```

**Comment by smizdev** [comment]
I too was seeing high CPU usage on an idle server, and around 2-3GB of idle mem:

![image](https://github.com/mornedhels/enshrouded-server/assets/1124388/132ee361-595e-4a4f-a4f1-fc0232e25ab5)

Tried the `dev-proton` image and the pod seems to get stuck after bootstrap and updating (cleared my prior server install on storage just to be sure)

```
2024-01-27 18:36:23.027 supervisord: enshrouded-bootstrap INFO - proton bootstrap finished
2024-01-27 18:36:23,272 INFO spawned: 'enshrouded-updater' with pid 106
2024-01-27 18:36:23,273 INFO success: enshrouded-updater entered RUNNING state, process has stayed up for > than 0 seconds (startsecs)
2024-01-27 18:36:23.275 supervisord: enshrouded-bootstrap enshrouded-updater: started
2024-01-27 18:36:23.294 supervisord: enshrouded-updater INFO - Running enshrouded-updater
2024-01-27 18:36:23.299 supervisord: enshrouded-bootstrap INFO - Bootstrap complete
2024-01-27 18:36:23,300 INFO exited: enshrouded-bootstrap (exit status 0; expected)
2024-01-27 18:36:23.897 supervisord: enshrouded-updater INFO - Enshrouded server is already the latest version
2024-01-27 18:36:24.123 supervisord: enshrouded-updater INFO - Enshrouded server is not running - starting
2024-01-27 18:36:24,329 INFO spawned: 'enshrouded-server' with pid 117
2024-01-27 18:36:34,346 INFO success: enshrouded-server entered RUNNING state, process has stayed up for > than 10 seconds (startsecs)
2024-01-27 18:36:34.348 supervisord: enshrouded-updater enshrouded-server: started
2024-01-27 18:36:34,376 INFO exited: enshrouded-updater (exit status 0; expected)
2024-01-27 18:36:34.375 supervisord: enshrouded-updater INFO - enshrouded-updater complete
```

Hope the move to proton helps. Will move back to the latest tag for the time being. Thanks for all the effort this is awesome!

**Comment by cp-fabian-pittroff** [comment]
@smizdev nice that it worked out. I will implement a fix when switching to the proton image.

I always get the window error, because I don't have setup the x server properly. But the enshrouded server doesn't really need a window. It falls back to console output anyways.

**Comment by zdegraaf** [comment]
Is this fix implemented in the current release? I've been able to play fine with only a single user that's on the same network as the server, but as soon as my friend, who's connecting remotely, tries to connect the performance tanks for both him and I. Sometimes he can't even get connected at all. 
The same as with the reported issue, CPU and memory usage remain low.

**Comment by cp-fabian-pittroff** [comment]
@zdegraaf there is a proton version available yes. But you have to specifically use the image tag (eg. `stable-proton`).
But that also doesn't solve all the problems. Enshrouded still needs a lot of cpu power to run smoothly.

**Comment by zdegraaf** [comment]
@cp-fabian-pittroff 
I tried running this using stable-proton and overall it was a better experience and was usable playing with two people. I still got the occasional yellow server overloaded message, but I didn't notice any performance drops or disruptive stuttering like I was getting with the Wine version. It still was only using a fraction of the resources I provided to it, though I didn't get a chance to note whether or not any single core. My environment for reference:
Hypervisor OS: Proxmox 6.4-13
Hypervisor CPU: AMD Threadripper 1950X 16 Cores
VM OS: Ubuntu 22.04.3 LTS
VM Specs: 12 cores, 16 GB Ram
Docker Version: Community 25.0.1

Thanks for the great container build, not just with good core features, but also with great stuff like the backup feature.
